### ISSUE TITLE
vagrant crashes when offline

### ISSUE THREAD

**Issue by lsh-0** [WhoopInc/vagrant-s3auth#26]
Vagrant can be used when offline if the box has already been downloaded. 

When using the s3auth plugin Vagrant crashes when offline attempting to resolve a DNS address.

Vagrant 1.7.4
Virtualbox 5.0.10
Arch Linux

also replicated on OSX El Capitan

Stacktrace below attempting to bring up a machine 'basebox-dev' using the image 'elifesciences/basebox'

```ruby
==> basebox-dev: Checking if box 'elifesciences/basebox' is up to date...
/opt/vagrant/embedded/lib/ruby/2.0.0/net/http.rb:878:in `initialize': getaddrinfo: Name or service not known (Seahorse::Client::NetworkingError)
	from /opt/vagrant/embedded/lib/ruby/2.0.0/net/http.rb:878:in `open'
	from /opt/vagrant/embedded/lib/ruby/2.0.0/net/http.rb:878:in `block in connect'
	from /opt/vagrant/embedded/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
	from /opt/vagrant/embedded/lib/ruby/2.0.0/net/http.rb:877:in `connect'
	from /opt/vagrant/embedded/lib/ruby/2.0.0/net/http.rb:862:in `do_start'
	from /opt/vagrant/embedded/lib/ruby/2.0.0/net/http.rb:857:in `start'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/seahorse/client/net_http/connection_pool.rb:292:in `start_session'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/seahorse/client/net_http/connection_pool.rb:104:in `session_for'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/seahorse/client/net_http/handler.rb:108:in `session'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/seahorse/client/net_http/handler.rb:60:in `transmit'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/seahorse/client/net_http/handler.rb:34:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/seahorse/client/plugins/content_length.rb:12:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/s3_request_signer.rb:149:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/s3_request_signer.rb:33:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/xml/error_handler.rb:8:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/s3_request_signer.rb:124:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/s3_redirects.rb:15:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/retry_errors.rb:87:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/retry_errors.rb:118:in `retry_request'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/retry_errors.rb:101:in `retry_if_possible'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/retry_errors.rb:89:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/retry_errors.rb:118:in `retry_request'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/retry_errors.rb:101:in `retry_if_possible'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/retry_errors.rb:89:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/retry_errors.rb:118:in `retry_request'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/retry_errors.rb:101:in `retry_if_possible'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/retry_errors.rb:89:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/s3_md5s.rb:33:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/s3_expect_100_continue.rb:21:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/s3_bucket_dns.rb:31:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/rest/handler.rb:7:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/user_agent.rb:12:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/s3_get_bucket_location_fix.rb:8:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/seahorse/client/plugins/endpoint.rb:41:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/param_validator.rb:21:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/seahorse/client/plugins/raise_response_errors.rb:14:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/s3_sse_cpk.rb:18:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/aws-sdk-core/plugins/param_converter.rb:20:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/seahorse/client/plugins/response_target.rb:21:in `call'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/seahorse/client/request.rb:70:in `send_request'
	from /home/luke/.vagrant.d/gems/gems/aws-sdk-core-2.1.33/lib/seahorse/client/base.rb:207:in `block (2 levels) in define_operation_methods'
	from /home/luke/.vagrant.d/gems/gems/vagrant-s3auth-1.2.0/lib/vagrant-s3auth/util.rb:69:in `get_bucket_region'
	from /home/luke/.vagrant.d/gems/gems/vagrant-s3auth-1.2.0/lib/vagrant-s3auth/util.rb:54:in `s3_object_for'
	from /home/luke/.vagrant.d/gems/gems/vagrant-s3auth-1.2.0/lib/vagrant-s3auth/extension/downloader.rb:33:in `s3auth_download'
	from /home/luke/.vagrant.d/gems/gems/vagrant-s3auth-1.2.0/lib/vagrant-s3auth/extension/downloader.rb:68:in `rescue in execute_curl_with_s3auth'
	from /home/luke/.vagrant.d/gems/gems/vagrant-s3auth-1.2.0/lib/vagrant-s3auth/extension/downloader.rb:63:in `execute_curl_with_s3auth'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/util/downloader.rb:134:in `download!'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/box.rb:131:in `load_metadata'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/box.rb:159:in `has_update?'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/box_check_outdated.rb:46:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builder.rb:116:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `block in run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/call.rb:53:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builder.rb:116:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `block in run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/call.rb:53:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builder.rb:116:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `block in run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:214:in `action_raw'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:191:in `block in action'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/environment.rb:516:in `lock'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:178:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:178:in `action'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run'
```

**Comment by benesch** [comment]
Ah, shoot! Good catch. This is [deep] in my queue—but a PR would be much appreciated!

Disabling box updating is the unsatisfying workaround:

```
config.vm.box_check_update = false
```

At least the box update checker is pretty useless, since it most commonly prints a warning to the user *as* it boots up the box, which is like the most useless time ever to tell the user they should `vagrant destroy && vagrant box update && vagrant up`.

**Comment by benesch** [comment]
Fix released in v1.3.0. Thanks for reporting!

**Comment by lsh-0** [comment]
Thanks for this Nikhil. I'll have a chance to test today. Apologies for the
lack of pull request - no ruby skills, no time.

On Tue, Jan 19, 2016 at 3:50 AM, Nikhil Benesch <notifications@github.com>
wrote:

> Fix released in v1.3.0. Thanks for reporting!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WhoopInc/vagrant-s3auth/issues/26#issuecomment-172730249>
> .
>



-- 

Luke Skibinski

Formally defined,
Extensively tested,
Cryptic, abstract, thing.

--

eLife Sciences Publications, Ltd is a limited liability non-profit
non-stock corporation incorporated in the State of Delaware, USA, with
company number 5030732, and is registered in the UK with company number
FC030576 and branch number BR015634 at the address First Floor, 24 Hills
Road, Cambridge CB2 1JP.
